src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingExtensions.cs

```diff
@@ -284,11 +284,22 @@ public static TextChange SimpleDiff(this TextChange textChange, string text)
 
         internal static IEnumerable<TextSpan> GetAnnotatedSpans(SyntaxNode node, SyntaxAnnotation annotation)
         {
-            foreach (var nodeOrToken in node.GetAnnotatedNodesAndTokens(annotation))
+            if (annotation == SyntaxAnnotation.ElasticAnnotation)
             {
-                var firstToken = nodeOrToken.IsNode ? nodeOrToken.AsNode()!.GetFirstToken(includeZeroWidth: true) : nodeOrToken.AsToken();
-                var lastToken = nodeOrToken.IsNode ? nodeOrToken.AsNode()!.GetLastToken(includeZeroWidth: true) : nodeOrToken.AsToken();
-                yield return GetSpan(firstToken, lastToken);
+                var tokens = node.GetAnnotatedTrivia(SyntaxAnnotation.ElasticAnnotation).Select(tr => tr.Token).Distinct();
+                return AggregateSpans(tokens.Select(t => GetElasticSpan(t)));
+            }
```

Comment:
curious why this cahnge?

Reply:
A bit of consolidation - same pattern was used in multiple places so I merged everything into one method.